plugins/modules/oneview_firmware_bundle.py

```diff
@@ -0,0 +1,133 @@
+              C(present) will ensure that the firmware bundle is at OneView.
+              C(add_signature) will add the compsig to hotfix.
+        choices: ['present', 'add_signature']
+        required: True
+        type: str
+    file_path:
+      description:
+        - The full path of a local file to be uploaded.
+      required: True
+      type: str
+
+extends_documentation_fragment:
+- hpe.oneview.oneview
+- hpe.oneview.oneview.validateetag
+- hpe.oneview.oneview.params
+'''
+
+EXAMPLES = '''
+- name: Upload the Firmware Bundle
+  oneview_firmware_bundle:
+    config: "{{ config_file_path }}"
+    state: present
+    file_path: "/home/user/Downloads/hp-firmware.rpm"
+
+- name: Upload the hotfix file
+  oneview_firmware_bundle:
+    config: "{{ config_file_path }}"
+    state: present
+    file_path: "/home/user/Downloads/hp-hotfix.zip"
+
+- name: Add the compsig file
+  oneview_firmware_bundle:
+    config: "{{ config_file_path }}"
+    state: add_signature
+    file_path: "/home/user/Downloads/hp-hotfix.compsig"
+'''
+
+RETURN = '''
+firmware_bundle:
+    description: Has the facts about the OneView firmware bundle.
+    returned: On state 'present'. Can be null.
+    type: dict
+'''
+
+from ansible_collections.hpe.oneview.plugins.module_utils.oneview import OneViewModule
+
+
+class FirmwareBundleModule(OneViewModule):
+    MSG_ADDED = 'Firmware Bundle or Hotfix added successfully.'
+    MSG_ALREADY_PRESENT = 'Firmware Bundle or Hotfix is already present.'
+    MSG_ADD_SIG = 'Added the compsig successfully.'
+    MSG_SIG_ALREADY_PRESENT = 'Signature already present.'
+    MSG_HOTFIX_ABSENT = 'Hotfix is not present.'
+    RESOURCE_FACT_NAME = 'firmware_bundles'
+
+    def __init__(self):
+        argument_spec = dict(state=dict(required=True, choices=['present', 'add_signature']),
+                             file_path=dict(required=True, type='str'))
+
+        super().__init__(additional_arg_spec=argument_spec, validate_etag_support=True)
+        self.resource_client = self.oneview_client.firmware_bundles
+
+    def execute_module(self):
+        file_path = self.module.params['file_path']
+        self.current_resource = self.resource_client.get_by_name(file_path)
+
+        if self.state == 'present':
+            return self.__present(file_path)
+        elif self.state == 'add_signature':
+            return self.__add_compsig(file_path)
+
+    def __present(self, file_path):
+        if not self.current_resource:
+            self.current_resource = self.resource_client.upload(file_path)
+            return dict(changed=True, msg=self.MSG_ADDED, ansible_facts=dict(firmware_bundle=self.current_resource))
+        else:
+            return dict(changed=False, msg=self.MSG_ALREADY_PRESENT, ansible_facts=dict(firmware_bundle=self.current_resource.data))
+
+    def __add_compsig(self, file_path):
+        if self.current_resource and self.current_resource.data.get('signatureFileRequired') is False:
```

Comment:
don't we need to check null condition for self.current_resource.data?

Reply:
Here ```self.current_resource``` is a return value of ```get_by_name``` method and if that ```self.current_resource``` is defined then it will definitely have ```data``` attribute inside it.

Reply:
okay